packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts

```diff
@@ -107,6 +107,12 @@ export interface IComboBoxProps extends ISelectableDroppableTextProps<IComboBox>
    * Add additional content below the callout list.
    */
   onRenderLowerContent?: IRenderFunction<IComboBoxProps>;
+
+  /**
+   * Close callout when user is typing in input (relies on AllowFreeform: true)
+   * @default false;
+   */
+  inputWithoutCallout?: boolean;

```

Comment:
You should accomplish this by adding a dismissMenu function on the comboBox to close the callout. That will allow you to get the behavior you are looking for and other authors who may want to dismiss the other menu for another reason will be able to do that as well.